fetch_teleop/src/joystick_teleop.cpp

```diff
@@ -567,6 +568,188 @@ class HeadTeleop : public TeleopComponent
+    }
+    else if (!button_body_frame_pressed && button_end_effector_frame_pressed)
+    {
+      if (button_roll_pitch_pressed)
+      {
+        if (init_point_ == 0)
+        {
+          roll_offset_ = joy->axes[axis_roll_];
+          pitch_offset_ = joy->axes[axis_pitch_];
+        }
+
+        desired_.twist.angular.x = (joy->axes[axis_roll_] - roll_offset_) * max_vel_roll_;
+        desired_.twist.angular.y = (joy->axes[axis_pitch_] - pitch_offset_) * max_vel_pitch_;
+
+        init_point_++;
+      }
+      else
+      {
+        desired_.twist.angular.x = 0.0;
+        desired_.twist.angular.y = 0.0;
+        init_point_ = 0;
+      }
+
+      ref_frame_ = "end_effector_frame";
+      desired_.twist.linear.x = joy->axes[axis_x_] * max_vel_x_;
+      desired_.twist.linear.y = joy->axes[axis_y_] * max_vel_y_;
+      desired_.twist.linear.z = joy->axes[axis_z_] * max_vel_z_;
+      desired_.twist.angular.z = joy->axes[axis_yaw_] * max_vel_yaw_;
+    }
+
+    last_.header.frame_id = ref_frame_;
+    return true;
+  }
+
+  virtual void publish(const ros::Duration& dt)
+  {
+    if (active_)
+    {
+      // Ramp commands based on acceleration limits
+      last_.twist.linear.x = integrate(desired_.twist.linear.x, last_.twist.linear.x, max_acc_x_, dt.toSec());
+      last_.twist.linear.y = integrate(desired_.twist.linear.y, last_.twist.linear.y, max_acc_y_, dt.toSec());
+      last_.twist.linear.z = integrate(desired_.twist.linear.z, last_.twist.linear.z, max_acc_z_, dt.toSec());
+      
+      last_.twist.angular.x = integrate(desired_.twist.angular.x, last_.twist.angular.x, max_acc_roll_, dt.toSec());
+      last_.twist.angular.y = integrate(desired_.twist.angular.y, last_.twist.angular.y, max_acc_pitch_, dt.toSec());
+      last_.twist.angular.z = integrate(desired_.twist.angular.z, last_.twist.angular.z, max_acc_yaw_, dt.toSec());
+
+      cmd_vel_pub_.publish(last_);
+    }
+  }
+  
+  virtual bool start()
+  {
+    active_ = true;
+    return active_;
+  }
+  
+
+  virtual bool stop()
+  {
+    // Publish stop message
+    last_ = desired_ = geometry_msgs::TwistStamped();
+    cmd_vel_pub_.publish(last_);
+    
+    active_ = false;
+    return active_;
+  }
+
+private:
+  
+  // Buttons from params
+  int axis_x_, axis_y_, axis_z_, axis_roll_, axis_pitch_, axis_yaw_;
+  int button_body_frame_, button_end_effector_frame_, button_roll_pitch_;
+  int init_point_;
+
+  // Limits from params
+  double max_vel_x_, max_vel_y_, max_vel_z_;
+  double max_vel_roll_, max_vel_pitch_, max_vel_yaw_;
+  double max_acc_x_, max_acc_y_, max_acc_z_;
+  double max_acc_roll_, max_acc_pitch_, max_acc_yaw_;
+  double prev_roll_, prev_pitch_;
+  double roll_offset_, pitch_offset_;
+
+  // Support for multiplexor between teleop and application base commands
+  bool use_mux_;
+  std::string prev_mux_topic_;
+  ros::ServiceClient mux_;
+
+  // Twist output, odometry feedback
+  ros::Publisher cmd_vel_pub_;
```

Comment:
this should really just be called command_pub_ or cmd_pub_ -- cmd_vel really is synonymous with "base control" in ROS